plugins/Tour/Widgets/GetEngagement.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+namespace Piwik\Plugins\Tour\Widgets;
+
+use Piwik\API\Request;
+use Piwik\Common;
+use Piwik\Plugins\Tour\Engagement\Levels;
+use Piwik\Plugins\Tour\Engagement\Challenges;
+use Piwik\Widget\Widget;
+use Piwik\Widget\WidgetConfig;
+use Piwik\Piwik;
+
+class GetEngagement extends Widget
+{
+    const NUM_CHALLENGES_PER_PAGE = 5;
+
+    /**
+     * @var Challenges
+     */
+    private $challenges;
+
+    /**
+     * Levels
+     */
+    private $levels;
+
+    public function __construct(Challenges $challenges, Levels $levels)
+    {
+        $this->challenges = $challenges;
+        $this->levels = $levels;
+    }
+
+    public static function configure(WidgetConfig $config)
+    {
+        $config->setCategoryId('About Matomo');
+        $config->setName(Piwik::translate('Tour_BecomeMatomoExpert'));
+        $config->setOrder(99);
+
+        if (!Piwik::hasUserSuperUserAccess()) {
+            $config->disable();
+        }
+    }
+
+    public function render()
+    {
+        Piwik::checkUserHasSuperUserAccess();
+
+        $challenges = Request::processRequest('Tour.getChallenges');
+        $level = Request::processRequest('Tour.getLevel');
```

Comment:
Should these have `$defaultRequest = []`? I don't think it will change anything since the api methods don't have any parameters, just wondering if it would be better form.

Reply:
yep, will add it 👍